pkg/actions/lua/storage/client.go

```diff
@@ -0,0 +1,26 @@
+package storage
+
+import "github.com/Shopify/go-lua"
+
+type Client interface {
+	GetObject(l *lua.State) int
+	PutObject(l *lua.State) int
+	DeleteObject(l *lua.State) int
+	ListObjects(l *lua.State) int
+	DeleteRecursive(l *lua.State) int
+}
+
+func InitStorageClient(l *lua.State, client Client) {
```

Comment:
IMO we don't need this abstraction, I fear it's too early and make things more complex. 

To make it easier to use with exporter like `export_delta_log` you can use the function param [write_object](https://github.com/treeverse/lakeFS/blob/master/pkg/actions/lua/lakefs/catalogexport/delta_exporter.lua#L45) and pass a wrapper function  for azure storage client `write_object: function(bucket, key, data)` it doesn't need to be on the SDK level. 

1. GCS is not included here
2. In S3/Azure the interface has 2 missing functions  (delete_recursive, list_objects missing)
3. `PutObject(l *lua.State) int` is not not matching in both cases (referring to implementation with comment `// Skipping first argument as it is the host arg (bucket which is irrelevant in Azure)` in azure that you added.). 
4. Those SDK's are not meant to be used by export hooks only, our users should be able to use those SDK independently for their own needs, the combination of them here doesn't seem friendly to user experience.